lib/multi.c

```diff
@@ -1117,11 +1125,16 @@ static CURLMcode Curl_multi_wait(struct Curl_multi *multi,
   nfds += extra_nfds; /* add the externally provided ones */
 
 #ifdef ENABLE_WAKEUP
+#ifdef USE_WINSOCK
+  if(use_wakeup) {
+#else
   if(use_wakeup && multi->wakeup_pair[0] != CURL_SOCKET_BAD) {
+#endif
     ++nfds;
```

Comment:
Is increasing nfds still required with the new approach? I don't think it is used with WinSock anymore, right?

Reply:
I think the whole if block can be put into #ifndef USE_WINSOCK.

Reply:
(Same thing as the other case; it's used to check if there's anything to wait for)

Reply:
Are you sure? It is set to 0 in line 1150 below and not used in between, because the NUM_POLLS_ON_STACK is not relevant for Windows.

Reply:
Nicely solved, now nfds is only calculated once on Windows!